lib/caesar.rb

```diff
@@ -1,6 +1,74 @@
+require "pry"
+
 class Caesar
-  attr_reader :key
 
   def initialize
+    @aord = "a".ord #97
+    @capAord = "A".ord #65
+    @zord = "z".ord #122
+    @capZord = "Z".ord #90
+  end
+
+  def eng_to_cipher(message, rotation, decipher = false)
+    #need to ignore spaces
+    words = message.split
+    new_words = words.map do |word|
+      cipher_word(word, rotation, decipher)
+    end
+    new_words.join(" ")
+  end #e2c
+
+  def cipher_word(message, rotation, decipher)
+    letters = message.split("")
+    letters.map do |letter|
+      letter_num = letter.ord
+      new_letter_num = letter_num - rotation
```

Comment:
This won't work for all cases. Your check_bad_encipher_chars method won't handle the case when a letter is rotated off the end of the lower case alphabet and into the upper case alphabet. For instance, "a" rotated by 15 will become 82 in your code which is "R", but your code won't catch that because it sees R as a valid letter. This is why it is best to implement the alphabet as an array of all the letters and not as a range of integers i.e. @aord, @zord, @capZord, @capAord. This is a good case when data structures, i.e. Arrays, are crucial to problem solving. Having an array helps us think about rotation as a circular rather than linear operation.